app/models/runner.rb

```diff
@@ -237,4 +242,13 @@ def request_new_id
       end
     end
   end
+
+  def lock_runner_for_permitted_execution_time!
+    # We reserve the runner for the permitted execution time plus one second to ensure that the runner is not
+    # released before the execution is finished. This is especially important for the case that a command execution
+    # timed out, but subsequent commands are scheduled (i.e., when calculating a score and swapping a runner).
+    reservation = execution_environment.permitted_execution_time.seconds + 1.second
```

Comment:
What is the 1 second based on? Is it based on your observation / experience or "some" smallest period?

Reply:
It is based on my experience. Usually, swapping a runner on my local system takes around 400ms, but I've also observed times up to 700ms. In production, I expect that the value is marginally higher, due to network latency. We can also go for 1.5 or 2 seconds if you like, below would be challenging. I can also add a comment, should I?

Reply:
Hmm...I'd tend to add a comment on where this comes from...

Reply:
Nice!